jcl/src/java.base/share/classes/java/lang/invoke/MethodHandleNatives.java

```diff
@@ -25,14 +25,22 @@
 
 package java.lang.invoke;
 
-/*
- * Stub class for compilation
- */
-
 class MethodHandleNatives {
 
 	static LinkageError mapLookupExceptionToError(ReflectiveOperationException roe) {
-		throw OpenJDKCompileStub.OpenJDKCompileStubThrowError();
+		String exMsg = roe.getMessage();
+		LinkageError linkageErr;
+		if (roe instanceof IllegalAccessException) {
+			linkageErr = new IllegalAccessError(exMsg);
+		} else if (roe instanceof NoSuchFieldException) {
+			linkageErr = new NoSuchFieldError(exMsg);
+		} else if (roe instanceof NoSuchMethodException) {
+			linkageErr = new NoSuchMethodError(exMsg);
+		} else {
+			linkageErr = new IncompatibleClassChangeError(exMsg);
+		}
+		linkageErr.initCause(roe.getCause());
```

Comment:
Should the cause be roe rather than its cause?

Reply:
My interpretation of this method `mapLookupExceptionToError` is to map (or convert) incoming `ReflectiveOperationException` to a `LinkageError`, hence this `LinkageError` wasn't caused by `ReflectiveOperationException` and they share same causes.
On the other hand, if `roe.getCause()` is `null` then `roe` can be the cause of `linkageErr` instead.